### DISCUSSION TITLE
Possible Player Physics Patches

### DISCUSSION THREAD

**Discussion by 0lhi** [Wohlstand/TheXTech#636] [Planning]
Bring up your ideas for updated/customized player physics in this thread!

**Comment by LoveBodhi** [comment]
- Jump Height
- Bounce Height
- Walking Speed
- Running Frame

Also enable a flag (e.g. Set a variable to a value) via Riivolution-like option to load defined physics at start

(Mainly Lua system?)

**Comment by ChristianSilvermoon** [comment]
I think at this point it may be better to leave it to the scripting system and provide a good player physics API.

I like the idea of TheXTech's scripting system having 'layers' with priority, that go in a sort of space like:
1. Level-side scripting (scripts in the level's directory only)
2. Episode-side scripting (scripts in the episode's root)
3. User-scripts (scripts in a user's script directory. Apply to any episode the user has enabled them on. Should come with a big warning notice about compatability, etc. This gives the player the power to customize the way they play to their heart's content)
4. Game Mode Scripts (Would need to be explicitly enabled, perhaps from some manner of menu. These might do things like let you play random levels from your miscellaneous episodes, or provide something in the vain of a limited life challenge)

Physics tweaks could be handled via the user's scripts in that case, since they may want to use it with various episodes, but the warning would be important because changing physics _could_ break some episodes

**Reply by LoveBodhi** [reply]
Maybe Core Engine's goal is different from Asset Pack (User-made)?

**Reply by LoveBodhi** [reply]
> Physics tweaks could be handled via the user's scripts in that case, since they may want to use it with various episodes, but the warning would be important because changing physics could break some episodes

This means it maybe outside of the scope for base engine. We could try to fork instead...

**Comment by LoveBodhi** [comment]
Is this no longer in the scope?

**Comment by LoveBodhi** [comment]
The Possible Option may be set multiplier for each fields, or Offset...
